File: problems-extend-adapt/2601-2700/2678_passengers-over-sixty/solution.py
from typing import List


class Solution:
    def countOverSixty(self, details: List[str]) -> int:
        # The age is the two-digit field at offsets 11-12; char-code
        # arithmetic decodes it without building a substring. The count is
        # at most len(details) <= 100, so plain ints are plenty.
        count = 0
        for record in details:
            age = (ord(record[11]) - ord("0")) * 10 + (ord(record[12]) - ord("0"))
            if age > 60:
                count += 1
        return count

File: problems-extend-adapt/2601-2700/2678_passengers-over-sixty/solution.cpp
class Solution {
  public:
    int countOverSixty(vector<string> &details) {
        // The age is the two-digit field at offsets 11-12; char-code
        // arithmetic decodes it without building a substring. The count is
        // at most details.size() <= 100, so int is plenty.
        int count = 0;
        for (const auto &record : details) {
            int age = (record[11] - '0') * 10 + (record[12] - '0');
            if (age > 60) {
                ++count;
            }
        }
        return count;
    }
};
